### PR TITLE
module: picard sortsam

### LINKED ISSUE #600: new module: PICARD/SORTSAM
I think it would be good to have a module for [PICARD/SORTSAM](https://bioconda.github.io/recipes/picard/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
Add picard SortSam module

## PR checklist

Closes #600 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/picard/sortsam/main.nf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    output:
+    tuple val(meta), path("*.sorted.bam"), emit: bam
+    path "*.version.txt"                 , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[Picard FilterSamReads] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
```
Comment (drpatelh) [inline modules/picard/sortsam/main.nf:35]:
```suggestion
        log.info '[Picard SortSam] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
```

File: modules/picard/sortsam/meta.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - bam:
+      type: file
+      description: BAM/CRAM/SAM file
+      pattern: "*.{bam,sam}"
+
```
Comment (drpatelh) [inline modules/picard/sortsam/meta.yml:25]: Add entry for `sort_order` here too

**Review by drpatelh — COMMENTED** [review]
Nice! Few changes.

File: modules/picard/sortsam/meta.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      description: File containing software version
+      pattern: "*.{version.txt}"
+  - bam:
+      type: file
+      description: Sorted BAM/CRAM/SAM file
+      pattern: "*sorted.{bam}"
+  - sort_order:
+      type: value
+      description: Picard sort order type
+      pattern: "unsorted|queryname|coordinate|duplicate|unknown"
```
Comment (drpatelh) [inline modules/picard/sortsam/meta.yml:43]: This should be moved to the `input` section

Comment (jfy133) [reply modules/picard/sortsam/meta.yml]: Already noticed :sweat_smile:

**Review by drpatelh — APPROVED** [review]
LGTM!

**Pull request merged**